micro_manager/micro_cpp_wrapper.py

```diff
@@ -0,0 +1,31 @@
+import ctypes
+import numpy as np
+
+class MicroSimulation:
+    def __init__(self, lib_path:str, sim_id:int):
+        if not lib_path.endswith(".so"):
+            lib_path += ".so"
+        self.MicroSimulation_lib = ctypes.CDLL(lib_path)
+        self.MicroSimulation_lib.MicroSimulation_solve.argtypes = [ctypes.c_void_p, ctypes.POINTER(ctypes.c_double), ctypes.POINTER(ctypes.c_double), ctypes.c_double]
+        # return type is a std::pair of float and numpy array
+        self.obj = self.MicroSimulation_lib.MicroSimulation_new(sim_id)
+
+    def initialize(self):
+        self.MicroSimulation_lib.MicroSimulation_initialize(self.obj)
+
+    def solve(self, macro_data, dt:float):
+        macro_scalar_data_ptr = ctypes.c_double(macro_data["macro-scalar-data"])
+        macro_vector_data_ptr = macro_data["macro-vector-data"].ctypes.data_as(ctypes.POINTER(ctypes.c_double))
```

Comment:
Currently only made for the solverdummy data. We have to extract the data from the config file and adjust the data that is passed to c++ accordingly.